app/controllers/user/shares_controller.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+class User::SharesController < User::BaseController
+  before_action :set_share, only: %i[show edit update destroy]
+
+  # GET /shares
+  def index
+    @shares = current_user.shares
+  end
+
+  # GET /shares/1
+  def show; end
+
+  # GET /shares/new
+  def new
+    # pre-fill any supplied params (e.g. birth_record if creating from the birth
+    # record page)
+    @share = if params.keys.include? 'share'
+               User::Share.new(share_params)
+             else
+               User::Share.new
+             end
+    @organisations = OrganisationUser.all
+  end
+
+  # GET /shares/1/edit
+  def edit; end
+
+  # POST /shares
+  def create
+    @share = User::Share.new(share_params)
+    # shares are always associated with the current user
+    @share.user = current_user
+    @share.recipient_type = OrganisationUser.name
+
+    respond_to do |format|
+      if @share.save
+        format.html { redirect_to @share, notice: 'Share was successfully created.' }
+      else
+        format.html { render :new }
+      end
+    end
+  end
+
+  # PATCH/PUT /shares/1
+  def update
+    respond_to do |format|
+      if @share.update(share_params)
+        format.html { redirect_to @share, notice: 'Share was successfully updated.' }
+      else
+        format.html { render :edit }
+      end
+    end
+  end
+
+  # DELETE /shares/1
+  def destroy
+    @share.destroy
```

Comment:
How about this sets a `revoked_at` field instead?

Reply:
not implemented yet, it's another ticket